compilation error with gcc 8.2.1

- OS version: e.g. Archlinux
 - Compiler version: e.g GCC 8.2.1
 - Package version: master

When compiling :
```shell
In file included from /home/sjturpc/sjtuRPCDAQ/build/DQM4HEP/DQMCore/src/Archiver.cc:12:
/home/sjturpc/sjtuRPCDAQ/build/DQM4HEP/DQMCore/include/dqm4hep/Archiver.h:38:20: error: ‘function’ in namespace ‘std’ does not name a template type
       typedef std::function<bool(MonitorElementPtr)> SelectorFunction;
                    ^~~~~~~~
/home/sjturpc/sjtuRPCDAQ/build/DQM4HEP/DQMCore/include/dqm4hep/Archiver.h:38:15: note: ‘std::function’ is defined in header ‘<functional>’; did you forget to ‘#include <functional>’?
/home/sjturpc/sjtuRPCDAQ/build/DQM4HEP/DQMCore/include/dqm4hep/Archiver.h:17:1:
+#include <functional>
```
and : 
```shell
In file included from /home/sjturpc/sjtuRPCDAQ/build/DQM4HEP/DQMCore/src/Handle.cc:12:
/home/sjturpc/sjtuRPCDAQ/build/DQM4HEP/DQMCore/include/dqm4hep/Handle.h:92:12: error: ‘function’ in namespace ‘std’ does not name a template type
       std::function<Handle(void)>  m_newFunction = {nullptr};
            ^~~~~~~~
/home/sjturpc/sjtuRPCDAQ/build/DQM4HEP/DQMCore/include/dqm4hep/Handle.h:92:7: note: ‘std::function’ is defined in header ‘<functional>’; did you forget to ‘#include <functional>’?
/home/sjturpc/sjtuRPCDAQ/build/DQM4HEP/DQMCore/include/dqm4hep/Handle.h:20:1:
+#include <functional>
```
Adding #include <functional> solved the issue

--- a/DQMCore/include/dqm4hep/Archiver.h
+++ b/DQMCore/include/dqm4hep/Archiver.h
@@ -11,6 +11,9 @@
 #ifndef DQM4HEP_ARCHIVER_H
 #define DQM4HEP_ARCHIVER_H
 
+// -- stl headers
+#include <functional>
+
 // -- dqm4hep headers
 #include "dqm4hep/Internal.h"
 #include "dqm4hep/StatusCodes.h"

--- a/DQMCore/include/dqm4hep/Handle.h
+++ b/DQMCore/include/dqm4hep/Handle.h
@@ -18,6 +18,7 @@
 #include <typeinfo>
 #include <typeindex>
 #include <memory>
+#include <functional>
 
 namespace dqm4hep {
   

--- a/DQMOnline/include/dqm4hep/OnlineElement.h
+++ b/DQMOnline/include/dqm4hep/OnlineElement.h
@@ -128,14 +128,14 @@ namespace dqm4hep {
        *
        *  @param  resetQtests whether to also reset the quality tests 
        */
-      virtual void reset(bool resetQtests = true);
-      
+      virtual void reset(bool resetQtests = true) override;
+
       /**
        *  @brief  Convert the monitor element to json
        *  
        *  @param  object the json object to receive
        */
-      virtual void toJson(core::json &object) const;
+      virtual void toJson(core::json &object) const override;
       
 // #if ROOT_VERSION_CODE >= ROOT_VERSION(6, 14, 0)
 //       /**
@@ -195,7 +195,7 @@ namespace dqm4hep {
        *
        *  @param  reports the list of quality test reports to receive
        */
-      core::StatusCode runQualityTests(core::QReportMap &reports);
+      core::StatusCode runQualityTests(core::QReportMap &reports) override;
 
       /** 
        *  @brief  Run a specific quality test
@@ -203,7 +203,7 @@ namespace dqm4hep {
        *  @param  name the quality test name to run
        *  @param  report the quality test report to receive
        */
-      core::StatusCode runQualityTest(const std::string &name, core::QReport &report);
+      core::StatusCode runQualityTest(const std::string &name, core::QReport &report) override;
 
     private:
       /// The run number

